CMakeLists.txt

```diff
@@ -1,7 +1,11 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.9 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.9...3.27 FATAL_ERROR)
```

Comment:
Perhaps a comment on the version cap is warranted.

Reply:
This is not a cap (even though it looks like one). This is stating that we support CMake 3.9 and accept the NEW policies introduced in all versions between 3.9 and 3.27. HOOMD will still configure with CMake >= 3.28, but will default to the OLD policies introduced in versions after 3.27.

Thew NEW policies in some newer versions make `FindPython` behave more like the old code. For example with the old policy CMP0094, CMake would prefer a homebrew installed Python 3.12 over a miniforge installed 3.11 - even when miniforge appears first on the PATH.